crypto/cms/cms_env.c

```diff
@@ -955,3 +1039,22 @@ int cms_pkey_get_ri_type(EVP_PKEY *pk)
     }
     return CMS_RECIPINFO_TRANS;
 }
+
+int cms_pkey_is_ri_type_supported(EVP_PKEY *pk, int ri_type)
+{
+    int supportedRiType;
+
+    if (pk->ameth != NULL && pk->ameth->pkey_ctrl != NULL) {
```

Comment:
Non legacy keys will not have pk->ameth == NULL, is this ok?

Reply:
Well, it could be not OK, but it is currently consistent with the code in pkcs7/ and cms/